simtools/simtel/simtel_events.py

```diff
@@ -1,194 +1,253 @@
+
+        isInEnergyRange = list(map(
+            lambda e: e > energyRange[0] and e < energyRange[1],
+            self.summaryEvents['energy']
+        ))
+        isInCoreRange = list(map(
+            lambda r: r < coreMax,
+            self.summaryEvents['r_core']
+        ))
+        return np.sum(np.array(isInEnergyRange) * np.array(isInCoreRange))
+
+    @u.quantity_input(coreMax=u.m)
+    def selectEvents(self, energyRange=None, coreMax=None):
+        '''
+        Select sim_telarray events within a certain energy range and core radius.
 
-        def _get_bins(hist, axis=0):
-            ax_str = 'x' if axis == 0 else 'y'
-            return np.linspace(
-                hist['lower_' + ax_str],
-                hist['upper_' + ax_str],
-                hist['n_bins_' + ax_str] + 1
-            )
-
-        def _get_ax_lim(hist, axis=0):
-            if np.sum(hist['data']) == 0:
-                return 0, 1
-
-            bins = _get_bins(hist, axis=axis)
-
-            if hist['data'].ndim == 1:
-                non_zero = np.where(hist['data'] != 0)
-            else:
-                marginal = np.sum(hist['data'], axis=axis)
-                non_zero = np.where(marginal != 0)
+        Parameters
+        ----------
+        energyRange: Tuple (len 2)
+            Max and min energy of energy range, e.g. energyRange=(100 * u.GeV, 10 * u.TeV)
+        coreMax: astropy.Quantity (distance)
+            Maximum core radius for selecting showers, e.g. coreMax=1000 * u.m
+
+        Returns
+        -------
+        list
+            List of events.
+        '''
+        energyRange = self._validateEnergyRange(energyRange)
+        coreMax = self._validateCoreMax(coreMax)
 
-            return bins[non_zero[0][0]], bins[non_zero[0][-1] + 1]
+        selectedEvents = list()
+        for file in self.inputFiles:
+            with SimTelFile(file) as f:
 
-        pdfPages = PdfPages(figName)
-        for hist in combinedHists:
+                for event in f:
+                    energy = event['mc_shower']['energy']
+                    x_core = event['mc_event']['xcore']
+                    y_core = event['mc_event']['ycore']
+                    r_core = math.sqrt(math.pow(x_core, 2) + math.pow(y_core, 2))
 
-            self._logger.debug('Processing: {}'.format(hist['title']))
+                    if energy < energyRange[0] or energy > energyRange[1]:
+                        continue
 
-            fig = plt.figure(figsize=(8, 6))
-            ax = plt.gca()
-            ax.set_title(hist['title'])
+                    if r_core > coreMax:
+                        continue
 
-            if hist['n_bins_y'] > 0:
-                # 2D histogram
+                    selectedEvents.append(event)
+        return selectedEvents
 
-                xlim = _get_ax_lim(hist, axis=0)
-                ylim = _get_ax_lim(hist, axis=1)
+    @u.quantity_input(coreMax=u.m)
+    def countSimulatedEvents(self, energyRange=None, coreMax=None):
```

Comment:
This function assumes a straightforward powerlaw simulation. You do not allow weights, simulation in bins of energy, different functions other than a powerlaw, etc. 

That is also true for the code radius. Now that we have the option of simulating events based on a distribution which depends on the energy, this function would give the wrong number.

It is very efficient, much better than actually counting events, but I would make it clear in the documentation and maybe even in the name of the function.

Reply:
I don't think now it's the time for us to think about it.
I can come back to this in the future and fix it. I need to understand how sim_telarray stores the info about the simulated spectrum.

Reply:
I definitely don't want to write a function that deals with everything I mentioned. Just make it clear in the documentation that those assumptions are made. Otherwise in the future it would be hard for us (or someone else) to find this as a source of misinformation. 

I would even change the name of the function to be less general, to make it even clearer, but that is less critical.

Reply:
I agree with adding something to the documentation (will do that).
But I prefer to keep the signature generic as it is. In the future we should deal with these possible non-power law spectra inside the function and the user won't need to know anything it.

Reply:
OK, no problem, documentation is enough then.